apps/api/src/app/invites/invites.controller.ts

```diff
@@ -30,17 +33,36 @@ export class InvitesController {
     @Post()
     @UseGuards(AuthGuard)
     @UseGuards(ThrottlerGuard)
-    @ApiExcludeEndpoint()
+    @ApiBody({ type: CreateInviteDto })
+    @ApiHeader({ name: "x-api-key", required: true })
+    @ApiCreatedResponse({ type: Invite })
+    @ApiOperation({
+        description: "Creates a new group invite with a unique code."
+    })
     async createInvite(
+        @Headers() headers: Headers,
         @Req() req: Request,
         @Body() dto: CreateInviteDto
-    ): Promise<string> {
-        const { code } = await this.invitesService.createInvite(
-            dto,
-            req.session.adminId
-        )
+    ): Promise<Invite> {
+        let invite: Invite
 
-        return code
+        const apiKey = headers["x-api-key"] as string
+
+        if (apiKey) {
+            invite = await this.invitesService.createInviteWithApiKey(
+                dto,
+                apiKey
+            )
+        } else if (req.session.adminId) {
+            invite = await this.invitesService.createInviteManually(
+                dto,
+                req.session.adminId
+            )
+        } else {
+            throw new NotImplementedException()
+        }
+
```

Comment:
Maybe this function should return the same thing the other one is returning? `Promise<Omit<Invite, "id">>`

And also do the same at the end?

```ts
        invite.groupName = invite.group.name
        invite.groupId = invite.group.id

        return mapEntity(invite)
```

What do you think?

Reply:
`groupId` and `groupName` seems a bit redundant since we already have the `group` field. However, I can add them back to the returned object to match the other function signature.

Reply:
Yes, agree. Then these could be removed from the other too right?

```ts
        invite.groupName = invite.group.name
        invite.groupId = invite.group.id
```

The idea is that both return the same info.

Reply:
yup, let me get rid of these than!

Reply:
now is ready for review!